### DISCUSSION TITLE
Exported HTML files : Automatically add an attribute to all <a> elements

### DISCUSSION THREAD

**Discussion by msameen99** [TiddlyWiki/TiddlyWiki5#6334] [Q&A]
Hi,

I'm exporting my TW5 to Static HTML files using Node.js.

Is there a way to add the attribute "rel=external" automatically to ALL <a> elements in the output HTML files?
(So the final <a> element will looks like : `<a rel=external href="xyz">My Link</a>`)

Regards

**Comment by Jermolene** [comment]
> Hi,
> 
> 
> 
> I'm exporting my TW5 to Static HTML files using Node.js.
> 
> 
> 
> Is there a way to add the attribute "rel=external" automatically to ALL <a> elements in the output HTML files?
> 
> (So the final <a> element will looks like : `<a rel=external href="xyz">My Link</a>`)
> 
> 
> 
> Regards

As you've probably seen, external links are currently marked up with `rel="noopener noreferrer"`. On reflection, we should probably revisit that: firstly to add `external`. Also`noreferrer` isn't appropriate for all links. `noopener` is probably reasonable for all external links.

The current implementation embeds those values within the external link parser which means that it would have be a global wiki that would hold across an entire wiki.

**Reply by msameen99** [reply]
Thanks a lot Jeremy for your reply,

> external links are currently marked up with rel="noopener noreferrer"

So, what about the internal links? I mean the `<a>` tags that point to the other tiddler's HTML files, is there a way to automatically add the `rel=external` attribute?

My use case: I'm using the "Sveltekit" with the output HTML files, and its current implementation intercepts the clicks on <a> elements ([check here](https://kit.svelte.dev/docs#anchor-options-rel-external)).

My Workaround is replacing all `"<a "` string by `"<a rel=external "` , which is working BUT I'm sure that at sometime it'll break something  (my other option is to use  [unified.js](https://unifiedjs.com/), but this need to add a new step in the workflow)


Regards
